dask/config.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import print_function, division, absolute_import
+
+from contextlib import contextmanager
+import os
+import sys
+
+from .compatibility import FileExistsError
+
+
+config_paths = [
+    '/etc/dask',
+    os.path.join(sys.prefix, 'etc', 'dask'),
+    os.path.join(os.path.expanduser('~'), '.config', 'dask'),
+    os.path.join(os.path.expanduser('~'), '.dask')
+]
+
+if 'DASK_CONFIG' in os.environ:
+    config_paths.append(os.environ['DASK_CONFIG'])
+
+
+def update(old, new):
+    """ Update a nested dictionary with values from another
+
+    This is like dict.update except that it smoothly merges nested values
+
+    This operates in-place and modifies old
+
+    Example
+    -------
+    >>> a = {'x': 1, 'y': {'a': 2}}
+    >>> b = {'y': {'b': 3}}
+    >>> update(a, b)
+    {'x': 1, 'y': {'a': 2, 'b': 3}}
+
+    See Also
+    --------
+    merge
+    """
+    for k, v in new.items():
+        if k not in old and type(v) is dict:
+            old[k] = {}
+
+        if type(v) is dict:
+            update(old[k], v)
+        else:
+            old[k] = v
+    return old
+
+
+def merge(*dicts):
+    """ Update a sequence of nested dictionaries
+
+    This prefers the values in the latter dictionaries to those in the former
+
+    Example
+    -------
+    >>> a = {'x': 1, 'y': {'a': 2}}
+    >>> b = {'y': {'b': 3}}
+    >>> merge(a, b)
+    {'x': 1, 'y': {'a': 2, 'b': 3}}
+
+    See Also
+    --------
+    update
+    """
+    result = {}
+    for d in dicts:
+        update(result, d)
+    return result
+
+
+def collect_yaml(paths=config_paths):
+    """ Collect configuration from yaml files
+
+    This searches through ``config_paths``, expands to find all yaml or json
+    files, and then parses each file.
+    """
+    # Find all paths
+    file_paths = []
+    for path in paths:
+        if os.path.exists(path):
+            if os.path.isdir(path):
+                file_paths.extend(sorted([
```

Comment:
Looks like a job for glob?

Reply:
I'm not sure how best to replace this code with glob. It seems more complex than glob's common use.

Reply:
It's also not clear to me what benefit we get from supporting multiple configuration formats. If yaml is the common one, then we already have to rely on a third-party library.